### PR TITLE
[MX-435] Refactor follows and saves

### PR DESCRIPTION
**Pull request by ds300**
The type of this PR is: **Enhancement**

This PR resolves https://artsyproduct.atlassian.net/browse/MX-435

### Description

This refactors the saves and follows page to use our custom sticky tab page framework. In the process it was necessary to make the sticky tab bar scrollable, and to fix some minor layout issues in StickyTabPageFlatList that I hadn't accounted for yet.

✨ bonus ✨ 

- Refresh the typography on all saves+follows pages
- Add a loading placeholder for the artworks tabs (the other ones seem less likely to come up)
- Delete unused code and unused test files

#### Scrollable tab bar

so-many-tabs

#### Artworks list placeholder

artwork-placehoders

#### Typography Refresh

<details>
<summary>Artists list</summary>
<br>

image

</details>

<details>
<summary>Shows list</summary>
<br>

image

</details>

<details>
<summary>Categories List</summary>
<br>

image

</details>
<details>
<summary>Zero State</summary>
<br>

image

</details>

#skip_new_tests

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>Missing Test Files:

- `src/lib/Scenes/Favorites/__tests__/Favorites-tests.tsx`

If these files are supposed to not exist, please update your PR body to include "#skip_new_tests".</td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: src/lib/Components/StickyTabPage/StickyTabPageTabBar.tsx [hunk 1]
```diff
@@ -1,98 +1,161 @@
...
+export function spring(node: Animated.Value, toValue: number) {
+  return Animated.spring(node, { toValue, useNativeDriver: true, bounciness: -7, speed: 13 })
+}
+
+const ActiveTabBorder: React.FC<{ tabLayouts: LayoutRectangle[]; activeTabIndex: number }> = ({
+  tabLayouts,
+  activeTabIndex,
+}) => {
+  // We resize this border using the `scaleX` transform property rather than the `width` property, to avoid running
+  // animations on the JS thread, so we need to set an initial, pre-transform span for the border.
```
Comment (MounirDhahri) [inline src/lib/Components/StickyTabPage/StickyTabPageTabBar.tsx:122]: cool trick

File: src/lib/utils/placeholders.tsx [hunk 2]
```diff
@@ -72,20 +72,33 @@ export const PlaceholderButton: React.FC<ViewStyle> = ({ ...props }) => {
...
+
 export const PlaceholderRaggedText: React.FC<{ numLines: number; seed?: number }> = ({ numLines, seed = 10 }) => {
   const lengths = useMemo(() => {
     // create our own little deterministic math.random() to avoid snapshot churn
-    let x = seed
-    function random() {
-      const y = Math.sin(x++) * 10000
-      return y - Math.floor(y)
-    }
+    const rng = new RandomNumberGenerator(seed)
```
Comment (MounirDhahri) [inline src/lib/utils/placeholders.tsx:95]: 👏

File: src/lib/Components/Lists/ShowItemRow.tsx [hunk 3]
```diff
@@ -34,19 +35,10 @@ export class ShowItemRow extends React.Component<Props, State> {
...
     isFollowedSaving: false,
   }
-  // @ts-ignore STRICTNESS_MIGRATION
-  isTapped: boolean
-
-  // @ts-ignore STRICTNESS_MIGRATION
-  handleTap(_slug, _internalID) {
-    this.isTapped = true
-    // @ts-ignore STRICTNESS_MIGRATION
+  handleTap = debounce((_slug: string, _internalID: string) => {
```
Comment (brainbicycle) [inline src/lib/Components/Lists/ShowItemRow.tsx:38]: 🎉

File: src/lib/Components/StickyTabPage/StickyTabPageTabBar.tsx [hunk 4]
```diff
@@ -1,98 +1,161 @@
...
+  useEffect(() => {
+    if (allTabLayoutsArePresent) {
+      // attempt to center the active tab
+      let left = 20
+      for (let i = 0; i < activeTabIndex.current; i++) {
+        left += tabLayouts[i]!.width
+      }
+      const center = left + tabLayouts[activeTabIndex.current]!.width / 2
+      const scrollLeft = center - screen.width / 2
+      scrollViewRef.current?.scrollTo({ x: scrollLeft })
```
Comment (brainbicycle) [inline src/lib/Components/StickyTabPage/StickyTabPageTabBar.tsx:37]: 📜

**Pull request merged**